src/lib_c/x86_64-linux-gnu/c/sys/limits.cr

```diff
@@ -0,0 +1,5 @@
+lib LibC
+  UID_MAX     = 0xffffffff_u32 # max value for a uid_t. BSD based shim this is not defined in gnu-clib or musl.
+  GID_MAX     = 0xffffffff_u32 # max value for a gid_t. BSD based shim this is not defined in gnu-clib or musl.
```

Comment:
Don't make up constants if they don't exist. I plan to have automatically generated bindings in the near future (after I get my GC to a good working state). This code will break.

Use aliases under a Crystal namespace with defaults for targets that don't define these values (taken from POSIX preferably).

Reply:
To make that clear: if it's not on a man page it shouldn't be in LibC.

Reply:
@ysbaddaden: Some OS's have it in their clib. Some don't. Its also different OS to OS. I will add a shim further up for this then. I was reading that most people who need this on linux don't even bother with a constant which seems very, very, dirty to me. 

@RX14: No OS has its entire clib documented. UID_MAX is not documented in the man pages of Darwin, or OpenBSD , yet is well known about and in limits.h or sys/limits.h.

Reply:
This is fixed. I removed the LibC limits from the OS's that don't have them and added a `Limits` module inside each of User and Group that sets it based on the OS.